src/hotspot/os_cpu/windows_x86/os_windows_x86.cpp

```diff
@@ -503,34 +422,14 @@ void os::print_register_info(outputStream *st, const void *context, int& continu
     CASE_PRINT_REG(13, "R13=", R13); break;
     CASE_PRINT_REG(14, "R14=", R14); break;
     CASE_PRINT_REG(15, "R15=", R15); break;
-#else
-    CASE_PRINT_REG(0, "EAX=", Eax); break;
-    CASE_PRINT_REG(1, "EBX=", Ebx); break;
-    CASE_PRINT_REG(2, "ECX=", Ecx); break;
-    CASE_PRINT_REG(3, "EDX=", Edx); break;
-    CASE_PRINT_REG(4, "ESP=", Esp); break;
-    CASE_PRINT_REG(5, "EBP=", Ebp); break;
-    CASE_PRINT_REG(6, "ESI=", Esi); break;
-    CASE_PRINT_REG(7, "EDI=", Edi); break;
-#endif // AMD64
     }
 # undef CASE_PRINT_REG
     ++n;
   }
 }
 
 extern "C" int SpinPause () {
-#ifdef AMD64
```

Comment:
Weird that SpinPause is not implemented on Win64, but oh well. This whole SpinPause mess should be arch-specific, not OS/Arch specific, probably.

Reply:
@shipilev There _is_ a way to implement SpinPause on Windows/x64 though, if support is really as simple as a single pause instruction. Should I help implement this separately (After this PR is integrated, to avoid conflicts)? Although, the way SpinPause can be implemented is honestly so simple and trivial that @magicus could simply replace the entire body of this SpinPause with it in this PR

Reply:
Submit a separate PR and implement this :) Pretty sure you'll get into some dark territories in Windows/AArch64, see how Linux/AArch64 does this. But honestly, this whole `extern "C"` mess should probably be cleaned up in favor of arch-specific stubs or something like that...

Reply:
Oh, I was thinking about Windows/x64, but I guess I can consider Windows/ARM64 too. I had a look at Linux/ARM64 actually, and it seems like it doesn't actually properly support SpinPause? It seems like it uses the overhead of a method call to "implement" SpinPause. I had a look at some example assembly that could potentially be used to implement it for Windows/ARM64, but I don't know if it's correct. If you want, we could continue this discussion elsewhere